agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -198,51 +267,67 @@ async def execute_function(self, func: Callable[..., Any], *args, **func_params)
+
             is_coroutine_function = inspect.iscoroutinefunction(func)
-            start_time = time.perf_counter()
 
             if is_coroutine_function:
                 result = await func(*args, **func_params["params"])
             else:
                 result = func(*args, **func_params["params"])
 
-            end_time = time.perf_counter()
-            latency = end_time - start_time
+            if token is not None:
+                # check that it doesn't affect the tracing.tree
+                trace = ag.tracing.dump_trace()
+
+                tracing_context.reset(token)
 
             if isinstance(result, Context):
                 save_context(result)
+
             if isinstance(result, Dict):
-                return FuncResponse(**result, latency=round(latency, 4))
-            if isinstance(result, str):
-                return FuncResponse(
-                    message=result, usage=None, cost=None, latency=round(latency, 4)
-                )
-            if isinstance(result, int) or isinstance(result, float):
-                return FuncResponse(
-                    message=str(result),
-                    usage=None,
-                    cost=None,
-                    latency=round(latency, 4),
-                )
-            if result is None:
-                return FuncResponse(
-                    message="Function executed successfully, but did return None. \n Are you sure you did not forget to return a value?",
-                    usage=None,
-                    cost=None,
-                    latency=round(latency, 4),
+                if "message" in result:
+                    data = {"message": result["message"]}
+                else:
+                    data = result
+            elif isinstance(result, str):
+                data = {"message": result}
+            elif isinstance(result, int) or isinstance(result, float):
+                data = {"message": str(result)}
+
+            if data is None:
+                warning = (
+                    "Function executed successfully, but did return None. \n Are you sure you did not forget to return a value?",
                 )
+
+                data = {"message": warning}
+
+            return BaseResponse(data=data, trace=trace)
+
         except Exception as e:
             self.handle_exception(e)
-        return FuncResponse(message="Unexpected error occurred when calling the @entrypoint decorated function", latency=0)  # type: ignore
 
     def handle_exception(self, e: Exception):
-        """Handle exceptions."""
+        try:
+            status_code = e.status_code if hasattr(e, "status_code") else 500
+            message = str(e)
+            stacktrace = traceback.format_exception(e, value=e, tb=e.__traceback__)  # type: ignore
+
+        except:
+            status_code = 500
+            message = ("Unexpected error occurred when calling @entrypoint or @route.",)
+            stacktrace = traceback.format_exc()
```

Comment:
I'm curious as to why we have this try-except block?

Reply:
The behavior was already there but without a `try-except`. I didn't want to remove it.
I'm not sure myself.

Reply:
That's alright. Let's leave it as it is.